torch_xla/experimental/tpu.py

```diff
@@ -94,18 +109,20 @@ def version() -> int:
   except requests.HTTPError as e:
     raise EnvironmentError('Failed to get TPU metadata') from e
 
-  match = re.match(r'^v(\d)-(\d+)$', env['ACCELERATOR_TYPE'])
+  match = re.match(r'^v(\d)-(\d+)$', env[xenv.ACCELERATOR_TYPE])
   return int(match.groups()[0])
 
 
 def get_worker_ips() -> List[str]:
   """Returns ordered list of TPU worker IPs from TPU metadata."""
-  metadata = _get_metadata('worker-network-endpoints')
-
-  # Workers have format 'hostname:uid:ip,hostname:uid:ip,...'
-  workers = metadata.split(',')
-  ips = [worker.split(':')[2] for worker in workers]
-
+  metadata = xu.getenv_as(xenv.TPU_WORKER_HOSTNAMES, str, '')
+  if metadata is '':
+    metadata = _get_metadata('worker-network-endpoints')
+    # Workers have format 'hostname:uid:ip,hostname:uid:ip,...'
+    workers = metadata.split(',')
+    ips = [worker.split(':')[2] for worker in workers]
+  else:
+    ips = metadata.split(',')
```

Comment:
super-nit: this is technically not a list of IPs anymore, since we may pass hostnames in as `TPU_WORKER_HOSTNAMES`. I suggest `hostnames` to be more general

Reply:
Done